src/google/protobuf/stubs/port.h

```diff
@@ -394,12 +395,10 @@ class Bits {
   static uint32 Log2FloorNonZero(uint32 n) {
 #if defined(__GNUC__)
   return 31 ^ static_cast<uint32>(__builtin_clz(n));
-#elif defined(COMPILER_MSVC) && defined(_M_IX86)
-  _asm {
-    bsr ebx, n
-    mov n, ebx
-  }
-  return n;
+#elif defined(_MSC_VER)
+  unsigned long where;
+  _BitScanReverse(&where, n);
```

Comment:
Is this more efficient than the original assembly?

Reply:
No, it's just as effective but works also for 64bit build (inline assembler can only be used in 32-bit compiler `_M_IX86`)